### PR TITLE
Reboot SwingSet on bulldozer upgrade

### PR DESCRIPTION
**Pull request by michaelfig**
refs: #6644

## Description

When upgrading to `agoric-upgrade-10`, SwingSet will be reinitialized.  This PR adds that initialization to the upgrade handler so that it completes during upgrade before any new blocks are produced.

### Security Considerations

Improves security:
- no user transactions can interpose until the upgrade is complete
- the chain will halt if there are errors during that upgrade, without committing to potentially corrupted state.

### Scaling Considerations

Avoids costly upgrade work impacting blocks beyond the upgrade block itself.

### CODE REVIEW

**Comment by dckc** [comment]
suggsest: refs #6644

File: golang/cosmos/app/app.go [hunk 1]
```diff
@@ -794,7 +794,21 @@ func upgrade10Handler(app *GaiaApp, targetUpgrade string) func(sdk.Context, upgr
 		app.VstorageKeeper.MigrateNoDataPlaceholders(ctx) // upgrade-10 only
 		normalizeProvisionAccount(ctx, app.AccountKeeper)
 
-		return app.mm.RunMigrations(ctx, app.configurator, fromVm)
+		mvm, err := app.mm.RunMigrations(ctx, app.configurator, fromVm)
```
Comment (JimLarson) [inline golang/cosmos/app/app.go:797]: Check if err is non-nil here, and if so return. We don't want to proceed with swingset initialization in this case.

File: golang/cosmos/app/app.go [hunk 2]
```diff
@@ -794,7 +794,21 @@ func upgrade10Handler(app *GaiaApp, targetUpgrade string) func(sdk.Context, upgr
...
+		// Just run the SwingSet kernel to finish bootstrap and get ready to open for
+		// business.
+		stdlog.Println("Rebooting SwingSet")
+		err2 := swingset.BootSwingset(ctx, app.SwingSetKeeper)
+
+		// fmt.Fprintf(os.Stderr, "BOOTSTRAP_BLOCK Returned from swingset: %s, %v\n", out, err)
+		if err2 != nil {
+			// NOTE: A failed BOOTSTRAP_BLOCK means that the SwingSet state is inconsistent.
+			// Panic here, in the hopes that a replay from scratch will fix the problem.
+			panic(err)
```
Comment (JimLarson) [inline golang/cosmos/app/app.go]: Should be `err2`, but if you return on error right after `RunMigrations()` above, you can just reuse `err` and this line will be correct.

**Review by JimLarson — APPROVED** [review]
LGTM with the changes below - except for the Dockerfile changes which I can't really speak to.

**Pull request merged**